design-proposals/sdn-ipam-secondary-nets.md

```diff
@@ -0,0 +1,450 @@
+# Overview
+
+## Motivation
+Virtual Machine owners want to offload IPAM from their custom solutions (e.g.
+custom DHCP server running on their cluster network) to SDN.
```

Comment:
> > Only secondary networks, as explicitly indicated on the goals (and non-goals) section.
> 
> When reading the motivation, I did not understand the actual need is for secondary networks. I guess you could explain why this is not needed for the pod network but is needed for the secondaries.

The pod network is managed / owned by kubernetes. A direct consequence of that is users get IPAM on it already. It's part of what it does. I.e. they don't have to manage IP addresses / configure DHCP servers / etc. The platform does that for them.

I hope this is clear enough.

> 
> > Well, for multiple ones.
> > That depends on the configuration provided by the cluster admin in the NAD (routes / DNS). It has nothing to do with this feature AFAIU.
> 
> The feature is triggered by a need. I think it is worth expressing if the need is for a single one or more. It may influence the solution proposed.

It isn't specified, but I assume asking for IPAM on secondary interfaces means just that ... IPAM on secondary interfaces. I.e. every secondary interface can have IPAM depending on the config set by the user.

> 
> > > How existing clusters resolve the configuration challenges will help (e.g. with a DHCP server, cloud-init config, scripts, etc).
> > 
> > 
> > I do not understand what you're asking for. Are you asking for anything here ?
> 
> Yes, I am asking to explain how the legacy VM or physical machines handled this need. That way, I can understand if you take a similar solution to Kubevirt or you invent a new solution to an old problem. Given a problem/challenge, how it was solved so far in other platforms (including bare-metal) can help provide context and support this proposal.

I still don't understand. Are you asking how other platforms implement IPAM, or are you asking how other platforms get around not having IPAM ? The response to the latter is stated right there in the motivation - they use DHCP servers, and static IP addressing.